request_test.go

```diff
@@ -352,31 +352,46 @@ func TestRequest(t *testing.T) {
 }
 
 func TestHandleHTTPHEADRequest(t *testing.T) {
-	modifierFunc := func(m *RequestModifier) {
-		m.Header.Set("foo", "bar")
-	}
-	w := utils.NewWriterStub()
-	req := &http.Request{Method: "HEAD", Header: make(http.Header)}
 	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		fmt.Fprint(w, "Hi")
 		st.Expect(t, r.Header.Get("foo"), "")
 	})
-	interceptor := Request(modifierFunc)
+	interceptor := Request(func(m *RequestModifier) {
+		m.Header.Set("foo", "bar")
+	})
+	w := utils.NewWriterStub()
+	req := &http.Request{Method: "HEAD", Header: make(http.Header)}
 	interceptor.HandleHTTP(w, req, handler)
 	st.Expect(t, string(w.Body), "Hi")
 }
 
 func TestHandleHTTPOPTIONSRequest(t *testing.T) {
```

Comment:
Yeah I agree. I thought about it when I was reviewing the PR but decided against because I didn't know what's the idiomatic way to write it in Go.